### PR TITLE
Vertex and p_parameter for Parabolas declared symbolically

### LINKED ISSUE #14461: Parabola vertex can not be determined if Parabola is declared symbolically.
```
>>> from sympy import *
>>> a = symbols('a')
>>> l = Line((-a, 0), slope=oo)
>>> p = Parabola((a, 0), l)
>>> p.vertex
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/geometry/parabola.py", line 412, in vertex
    vertex = Point(focus.args[0] - self.p_parameter, focus.args[1])
  File "sympy/geometry/parabola.py", line 374, in p_parameter
    if (x < self.focus.args[0]):
  File "sympy/core/relational.py", line 229, in __nonzero__
    raise TypeError("cannot determine truth value of Relational")
TypeError: cannot determine truth value of Relational
```

### PR DESCRIPTION
**Pull request by jsbarbosa**
#### References to other Issues or PRs

Fixes #14461

#### Brief description of what is fixed or changed
Before, to define the sign of the `p_parameter`, a comparison between the focus points and the directrix coefficients was made. Now, the sign of the `p_parameter` is obtained using the `sign` function. 

#### Release Notes

* geometry
  * Fix `p_parameter` for Parabolas with symbolic variables

### CODE REVIEW

**Comment by sidhantnagpal** [comment]
Can you add the test, referenced in the issue, under [test_parabola.py](https://github.com/sympy/sympy/blob/master/sympy/geometry/tests/test_parabola.py)?

File: sympy/geometry/parabola.py [hunk 1]
```diff
@@ -369,19 +369,12 @@ def p_parameter(self):
 
         """
         if (self.axis_of_symmetry.slope == 0):
```
Comment (sidhantnagpal) [inline sympy/geometry/parabola.py]: The parentheses could be dropped here (can be included in this PR).

**Comment by sidhantnagpal** [comment]
Aside from the couple of comments, this looks good to me.

**Comment by jsbarbosa** [comment]
I have added a simple test that uses symbols with no assumptions. I also cleared the parenthesis in `p_parameter`.

File: sympy/geometry/tests/test_parabola.py [hunk 2]
```diff
@@ -62,6 +68,9 @@ def test_parabola_geom():
     assert pa8.p_parameter == pa9.p_parameter
     assert pa8.vertex == pa9.vertex
     assert pa8.equation() == pa9.equation()
+    assert pa10.focal_length == sqrt((a - b) ** 2) / 2 # if a, b real == abs(a - b)/2
+    assert pa10.focal_length == pa11.focal_length
+    assert Point2D(*pa10.vertex[::-1]) == pa11.vertex # change axis x->y, y->x on pa10
```
Comment (sidhantnagpal) [inline sympy/geometry/tests/test_parabola.py]: It'd be also good to include what it actually equals to.

```suggestion
    assert pa11.vertex == Point(*pa10.vertex[::-1]) == Point(a, 
                            a - sqrt((a - b)**2)*sign(a - b)/2) # change axis x->y, y->x on pa10
```

File: sympy/geometry/tests/test_parabola.py [hunk 3]
```diff
@@ -62,6 +68,9 @@ def test_parabola_geom():
     assert pa8.p_parameter == pa9.p_parameter
     assert pa8.vertex == pa9.vertex
     assert pa8.equation() == pa9.equation()
+    assert pa10.focal_length == sqrt((a - b) ** 2) / 2 # if a, b real == abs(a - b)/2
+    assert pa10.focal_length == pa11.focal_length
```
Comment (sidhantnagpal) [inline sympy/geometry/tests/test_parabola.py]: Or just `pa10.focal_length == pa11.focal_length == sqrt((a - b)**2)/2`.

**Comment by sidhantnagpal** [comment]
Looks good. Thanks for contributing.

**Pull request merged**